internal/handlers/common/update_array_operators.go

```diff
@@ -414,3 +417,69 @@ func processPullAllArrayUpdateExpression(doc, update *types.Document) (bool, err
 
 	return changed, nil
 }
+
+// processPullArrayUpdateExpression changes document according to $pull array update operator.
+// If the document was changed it returns true.
+func processPullArrayUpdateExpression(doc *types.Document, update *types.Document) (bool, error) {
+	var changed bool
+
+	iter := update.Iterator()
+	defer iter.Close()
+
+	for {
+		key, pullValueRaw, err := iter.Next()
+		if err != nil {
+			if errors.Is(err, iterator.ErrIteratorDone) {
+				break
+			}
+
+			return false, lazyerrors.Error(err)
+		}
+
+		path, err := types.NewPathFromString(key)
+		if err != nil {
+			return false, lazyerrors.Error(err)
+		}
+
+		// If the path does not exist, try to set it.
+		// If it is not possible to set the path return error.
+		// This will deal with cases like $pull on a non-existing field or unset field.
+		if !doc.HasByPath(path) {
+			if err = doc.SetByPath(path, types.MakeArray(1)); err != nil {
+				return false, commonerrors.NewWriteErrorMsg(
+					commonerrors.ErrUnsuitableValueType,
+					err.Error(),
+				)
+			}
```

Comment:
Still, can we return `false, nil` somewhere here? :) If `HasByPath` gave us `false`, and `SetByPath` didn't give us error, do we need to navigate through the rest of the code?

Reply:
I think it's possible to set `continue` but I think `false, nil` would skip the next value of the iterator and that's not what we want.

Reply:
Ah! Yes, you are right!